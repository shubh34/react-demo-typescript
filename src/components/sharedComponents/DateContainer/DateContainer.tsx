import React from 'react';
import { Moment } from 'moment';
import './DateContainer.css';

interface IDateContainer {
	date?: string | Moment;
	className?: string;
}
const DateContainer: React.FC<IDateContainer> = ({ date, className = '' }) => {
	return (
		<div className='date_container'>
			<label className='date_container_label'>
				<span className='date_container_meta'>Date: </span>
				<strong>{date}</strong>
			</label>
		</div>
	);
};

export default DateContainer;
