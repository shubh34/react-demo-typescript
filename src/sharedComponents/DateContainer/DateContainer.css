.date_container {
    min-width: 150px;
}
.date_container_label {
    line-height: 30px;
}
@media only screen and (min-width: 768px) {
    .date_container_meta {
        display: none;
    }
  }
