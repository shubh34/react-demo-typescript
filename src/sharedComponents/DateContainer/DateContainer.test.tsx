import React from 'react';
import { shallow } from 'enzyme';
import DateContainer from './DateContainer';

it('renders without crashing', () => {
	const div = shallow(<DateContainer date='23 Jan 1992' />);
	expect(div).toMatchSnapshot();
});
