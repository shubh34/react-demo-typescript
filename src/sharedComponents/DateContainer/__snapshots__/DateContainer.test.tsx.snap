// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="date_container"
>
  <label
    className="date_container_label"
  >
    <span
      className="date_container_meta"
    >
      Date: 
    </span>
    <strong>
      23 Jan 1992
    </strong>
  </label>
</div>
`;
